pkg/kubectl/cmd/apply.go

```diff
@@ -207,14 +223,25 @@ func parsePruneResources(mapper meta.RESTMapper, gvks []string) ([]pruneResource
 	return pruneResources, nil
 }
 
-func RunApply(f cmdutil.Factory, cmd *cobra.Command, out, errOut io.Writer, options *ApplyOptions) error {
+func (o *ApplyOptions) Complete(f cmdutil.Factory, cmd *cobra.Command) error {
+	o.RecordFlags.Complete(cmd, f.Command(cmd, false))
+	recorder, err := o.RecordFlags.ToRecorder()
+	if err != nil {
+		return err
+	}
+	o.Recorder = recorder
+
+	return nil
+}
+
+func (o *ApplyOptions) Run(f cmdutil.Factory, cmd *cobra.Command) error {
```

Comment:
make and merge this change now with TODO that says "stop passing factory and cmd"